warp10/src/main/java/io/warp10/script/aggregator/Percentile.java

```diff
@@ -57,26 +57,33 @@ public Builder(String name, boolean forbidNulls) {
 
     @Override
     public Object apply(WarpScriptStack stack) throws WarpScriptException {
-      Object value = stack.pop();
+      Object top = stack.pop();
 
-      if (!(value instanceof Number)) {
+      boolean interpolate = false;
```

Comment:
The naming of this variable is confusing. The behaviour with interpolate = false is a type of interpolation.

Reply:
I disagree. Even if it isn't the term used by Hyndman (discontinuous and continuous is), interpolation make the behavior very clear: we're interpolating between GTS points or not. This is equally true for users which may not be experts in percentile but want to choose between a point in the original GTS or an interpolated point.